src/pool.jl

```diff
@@ -88,24 +88,40 @@ Base.getindex(pool::CategoricalPool, i::Integer) = pool.valindex[i]
 Base.get(pool::CategoricalPool, level::Any) = pool.invindex[level]
 Base.get(pool::CategoricalPool, level::Any, default::Any) = get(pool.invindex, level, default)
 
-@inline function Base.get!(pool::CategoricalPool{T, R}, level::Any) where {T, R}
-    get!(pool.invindex, level) do
-        x = convert(T, level)
-        n = length(pool)
-        if n >= typemax(R)
-            throw(LevelsException{T, R}([level]))
+@inline function push_get!(pool::CategoricalPool{T, R}, level) where {T, R}
+    x = convert(T, level)
+    n = length(pool)
+    if n >= typemax(R)
+        throw(LevelsException{T, R}([level]))
+    end
+
+    i = R(n + 1)
+    push!(pool.index, x)
+    push!(pool.order, i)
+    push!(pool.levels, x)
+    push!(pool.valindex, catvalue(i, pool))
+    i
+end
+
+@inline function Base.get!(pool::CategoricalPool, level::Any)
+    v = get!(pool.invindex, level) do
+        if isordered(pool)
+            throw(ArgumentError("'$level' is a new level but the CategoricalArray uses ordered levels and ordered levels can't be extended implicitly, use function `levels!` to set new levels and include the new level"))
```

Comment:
I'd rather say ``"cannot add new level $level since ordered pools cannot be extended implicitly. Use the levels! function to set new levels, or the ordered! function to mark the pool as unordered."``. Currently other error messages don't use quotes, we should probably change this everywhere, but better stay consistent.

Also try to break the line where possible, cf. https://github.com/gustafsson/CategoricalArrays.jl/blob/e2b6d9508db99db8807e32b1611c9b425c99973a/src/array.jl#L568

Finally, this should probably be a special exception type (e.g. `OrderedLevelsException`) so that this situation can be caught. See how `LevelsException` works. (If that's too much work you can leave this as-is.)